docs/guides/class-maps.md

```diff
@@ -0,0 +1,192 @@
+
+```php
+use BookAttachment;
+use Timber\Attachment;
+
+add_filter( 'timber/post/classmap', function() {
+    return [
+        'attachment' => function( \WP_Post $post ) {
+            if ( 'book' === get_post_type( $post->post_parent ) {
+                return BookAttachment::class;
+            }
+            
+            return Attachment::class;
+        },
+    ];
+} );
+```
+
+## The Term Class Map
+
+When you extend `Timber\Term`, the logic you add is usually for a certain taxonomy. With the `timber/term/classmap` filter, you can tell Timber which class it should use for certain taxonomies.
+
+The Term Class Map is used:
+
+- When you get a post through `Timber::get_term()`.
+- When you get a collection of posts through `Timber::get_terms()`.
+- When you use a function that returns other terms, like `$post->terms()`.
+
+**functions.php**
+
+```php
+use Genre;
+
+add_filter( 'timber/term/classmap', function() {
+    return [
+        'genre' => Genre::class,
+    ];
+} );
+```
+
+Taxonomies that you don’t list in the Term Class Map will take the default `Timber\Term` class.
+
+When you need more fine-grained control over which class is used for your term object, you can use a callback function:
+
+```php
+use ComedyGenre;
+use Genre;
+
+add_filter( 'timber/term/classmap', function() {
+    return [
+        'genre' => function( \WP_Term $term ) {
+            if ( $term->term_id === 2 ) {
+                return ComedyGenre::class;
+            }
+            
+            return Genre::class;
+        },
+    ];
+} );
+```
+
+The callback function receives a `WP_Term` object and should return the name of the class to use.
+
+## The Comment Class Map
+
+When you extend `Timber\Comment`, the logic you add is usually for comments related to a certain post type. With the `timber/comment/classmap` filter, you can tell Timber which class it should use for comments that belong to a certain post type.
+
+The Comment Class Map is used:
+
+- When you get a single comment through `Timber::get_comment()`.
+- When you get the comments for a post through `$post->comments()`.
+
+**functions.php**
+
+```php
+use CommentPost;
+use CommentBook;
+
+add_filter( 'timber/comment/classmap', function() {
+    return [
+        'post' => CommentPost::class,
+        'book' => CommentBook::class,
+    ];
+} );
+```
+
+Comments for post types that you don’t list in the Comment Class Map will take the default `Timber\Comment` class.
+
+When you need more fine-grained control over which class is used for your comment object, you can use a callback function:
+
+```php
+use BookComment;
+use BookChildComment;
+
+add_filter( 'timber/comment/classmap', function() {
+    return [
+        'book' => function( \WP_Post $post ) {
+            if ( 0 !== $post->post_parent ) {
+                return BookChildComment::class;
+            }
+            
+            return BookComment::class;
+        },
```

Comment:
@gchtr I think this callback should still get the `WP_Comment` object, rather than the associated post. Otherwise, if someone wants logic based on comment metadata or something, there's no way to do it because the comment ID is no longer in scope. If folks want logic based on the associated post, on the other hand, they can always just call `get_post($comment->comment_post_ID)`.

Reply:
@acobster  Thanks, that’s a very good input! I updated the guide in https://github.com/timber/timber/pull/2073/commits/4012fd625d5b951efb741ef02ba009b4b3e032cd.

When calling the filter, do you think we already have the associated post and could pass it as a second parameter, or do we only have the post type of the associated post?

Would (and should) this be possible?

```php
'book' => function( \WP_Comment $comment, \WP_Post $post ) {

}
```

Or even:

```php
'book' => function( \WP_Comment $comment, \Timber\Post $post ) {

}
```